src/main.rs

```diff
@@ -69,16 +76,18 @@ fn main() {
     // Parse & Build an AST
     let mut parser = Parser::new(lexer);
 
-    let ast_root = match parser.parse() {
+    let mut ast_root = match parser.parse() {
```

Comment:
mmmm, i feel like this tree should be immutable, and we should generate a new tree (e.g. IL), but idk